# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: security/manager/ssl/nsITLSSocketControl.idl
# Commit: b45bd4ea0000
# Full Hash: b45bd4ea0000e64e4b159cac01d09519d73166f7
# Author: Dana Keeler <dkeeler@mozilla.com>
# Date: 2022-11-17 09:39:01
# Regressor Bug: 1791633
# File Overlap Count: 1
# Description:
#   Bug 1791633 - separate nsITLSSocketControl from nsITransportSecurityInfo r=necko-reviewers,kershaw,jschanck
#   
#   Depends on D160311
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D160313
# ==============================================================================

diff -r 705439f1d269 -r b45bd4ea0000 security/manager/ssl/nsITLSSocketControl.idl
--- a/security/manager/ssl/nsITLSSocketControl.idl	Wed Nov 16 16:37:29 2022 +0000
+++ b/security/manager/ssl/nsITLSSocketControl.idl	Wed Nov 16 16:37:29 2022 +0000
@@ -7,8 +7,9 @@
 #include "nsISupports.idl"
 
 interface nsIInterfaceRequestor;
+interface nsITlsHandshakeCallbackListener;
+interface nsITransportSecurityInfo;
 interface nsIX509Cert;
-interface nsITlsHandshakeCallbackListener;
 
 %{C++
 #include "nsStringFwd.h"
@@ -16,10 +17,12 @@
 %}
 [ref] native nsCStringTArrayRef(nsTArray<nsCString>);
 
+// An interface describing an object that controls and holds information about
+// a TLS handshake.
+// NB: The implementations of this interface may only be used on the socket
+// thread (except for asyncGetSecurityInfo);
 [scriptable, builtinclass, uuid(418265c8-654e-4fbb-ba62-4eed27de1f03)]
 interface nsITLSSocketControl : nsISupports {
-    attribute nsIInterfaceRequestor     notificationCallbacks;
-
     void proxyStartSSL();
     void StartTLS();
 
@@ -92,11 +95,6 @@
      */
     readonly attribute uint32_t providerFlags;
 
-    /*
-     * The original TLS flags from the socket provider.
-     */
-    readonly attribute uint32_t providerTlsFlags;
-
     /* These values are defined by TLS. */
     const short SSL_VERSION_3   = 0x0300;
     const short TLS_VERSION_1   = 0x0301;
@@ -162,4 +160,23 @@
     [noscript] void disableEarlyData();
 
     [noscript] void setHandshakeCallbackListener(in nsITlsHandshakeCallbackListener callback);
+
+    /**
+     * The id used to uniquely identify the connection to the peer.
+     */
+    readonly attribute ACString peerId;
+
+    /**
+     * The securityInfo of the TLS handshake.
+     */
+    readonly attribute nsITransportSecurityInfo securityInfo;
+
+    /**
+     * Asynchronously obtain the securityInfo of the TLS handshake. Resolves
+     * with an nsITransportSecurityInfo. This should probably only be used in
+     * tests, where JS running on the main thread cannot access any of the
+     * other fields of nsITLSSocketControl.
+     */
+    [implicit_jscontext, must_use]
+    Promise asyncGetSecurityInfo();
 };